pkg/util/tracing/crdbspan.go

```diff
@@ -253,15 +253,18 @@ func (s *crdbSpan) getRecordingLocked(m mode) tracingpb.RecordedSpan {
 		rs.Tags[k] = v
 	}
 
+	if rs.Duration == -1 {
+		// -1 indicates an unfinished Span. For a recording it's better to put some
+		// duration in it, otherwise tools get confused. For example, we export
+		// recordings to Jaeger, and spans with a zero duration don't look nice.
+		rs.Duration = timeutil.Now().Sub(rs.StartTime)
+	}
+
 	// When nobody is configured to see our spans, skip some allocations
 	// related to Span UX improvements.
 	onlyBackgroundTracing := m == modeBackground && s.recordingType() == RecordingOff
 	if !onlyBackgroundTracing {
 		if rs.Duration == -1 {
-			// -1 indicates an unfinished Span. For a recording it's better to put some
-			// duration in it, otherwise tools get confused. For example, we export
-			// recordings to Jaeger, and spans with a zero duration don't look nice.
-			rs.Duration = timeutil.Now().Sub(rs.StartTime)
 			addTag("_unfinished", "1")
```

Comment:
@irfansharif — I believe this if block will never be entered now (there will no longer be any more `rs.Duration==1` as they have all been changed to current town - start time, so the "_unfinished" tag will never be added. Should the `addTag("_unfinished", "1")` be moved to line 261?

Reply:
Or are tags only added when `onlyBackgroundTracing` is off? In that case I can move the earlier logic where the duration is changed from -1 to the current duration to just after this block.

Reply:
Failing test answered my question, it does need to be moved to just after!